### ISSUE TITLE
Assertion failure in fstmakecontextsyms utility

### ISSUE THREAD

**Issue by dorozhkin-stc** [kaldi-asr/kaldi#3134]
The usage example provided in the code comment at [fstmakecontextsym.cc](https://github.com/kaldi-asr/kaldi/blob/8d60ee3d8b29b89fd2e4e800afde1627bcb6fecf/src/fstbin/fstmakecontextsyms.cc#L34) doesn't work giving an assertion failure:
```
$ fstmakecontextsyms phones.txt ilabels.int > context_syms.txt
fstmakecontextsyms phones.txt ilabels.int
ASSERTION_FAILED (fstmakecontextsyms[5.5.246-7a1908]:CreateILabelInfoSymbolTable():context-fst.cc:348) Assertion failed: (!info.empty() && !info[0].empty())

[ Stack-Trace: ]
kaldi::MessageLogger::LogMessage() const
kaldi::KaldiAssertFailure_(char const*, char const*, int, char const*)
fst::CreateILabelInfoSymbolTable(std::vector<std::vector<int, std::allocator<int> >, std::allocator<std::vector<int, std::allocator<int> > > > const&, fst::SymbolTable const&, std::string, std::string)
main
__libc_start_main
fstmakecontextsyms() [0x4034e9]

Aborted
```

OTH, according the online [documentation](http://kaldi-asr.org/doc/tree_externals.html#tree_ilabel), the first item of the ilabel_info object is supposed to be an empty vector. Just inverting the check `!info[0].empty()` seems to fix the issue. Here is the commit in my branch for the reference https://github.com/dorozhkin-stc/kaldi/commit/0afbadb25883f0f46f2f05dcb54f9d142e5e9d3e.

**Comment by danpovey** [comment]
Can you please make a PR for that?
We don't often run that code so I guess I can just about believe this was a bug (although I'm surprised it was never encountered before).

**Comment by danpovey** [comment]
sorry, my mouse is acting up.

**Comment by dorozhkin-stc** [comment]
OK, this is my PR: https://github.com/kaldi-asr/kaldi/pull/3142
CI has failed to build
```
kaldi-online.a(online-audio-source.o): In function `kaldi::OnlinePaSource::Callback(void const*, void*, long, PaStreamCallbackTimeInfo const*, unsigned long)':
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:161: undefined reference to `PaUtil_GetRingBufferWriteAvailable'
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:164: undefined reference to `PaUtil_WriteRingBuffer'
kaldi-online.a(online-audio-source.o): In function `OnlinePaSource':
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:71: undefined reference to `PaUtil_InitializeRingBuffer'
kaldi-online.a(online-audio-source.o): In function `kaldi::OnlinePaSource::Read(kaldi::Vector<float>*)':
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:122: undefined reference to `PaUtil_GetRingBufferReadAvailable'
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:122: undefined reference to `PaUtil_GetRingBufferReadAvailable'
/home/travis/build/kaldi-asr/kaldi/src/online/online-audio-source.cc:138: undefined reference to `PaUtil_ReadRingBuffer'
```
but I guess this is not my fault.

**Comment by jtrmal** [comment]
no, it's not your fault.
that's coming from a compilation of different stuff.
y.

**Comment by kkm000** [comment]
Fixed by #3142.
